cmd/juju/controller/register.go

```diff
@@ -623,3 +632,40 @@ type byteAtATimeReader struct {
 func (r byteAtATimeReader) Read(out []byte) (int, error) {
 	return r.Reader.Read(out[:1])
 }
+
+// ensureNotKnownEndpoint checks whether any controllers in the local client
+// cache contain the provided endpoint and returns an error if that is the
+// case.
+func ensureNotKnownEndpoint(store jujuclient.ClientStore, endpoint string) error {
+	existingDetails, existingName, err := store.ControllerByAPIEndpoints(endpoint)
+	if err != nil && !errors.IsNotFound(err) {
+		return errors.Trace(err)
+	}
+
+	if existingDetails == nil {
```

Comment:
I wonder if it would be good to also do this check?

```suggestion
	if existingDetails == nil || existingName == "" {
```

Reply:
If no local controller with this endpoint exists, L640 (`store.ControllerByAPIEndpoints`) will return a NotFound error and details will be nil (existingName will be blank too). So we don't need the extra check here. I don't think you can have a nameless controller, can you?

Reply:
I didn't know hence why i thought the extra test might be worth doing, but if we can't get coverage then it's pointless having it.